File: library/src/main/java/org/fbme/lib/st/types/GenericType.java
package org.fbme.lib.st.types;

public enum GenericType implements DataType {
    ANY_BIT,
    ANY_DATE,
    ANY_DERIVED,
    ANY_ELEMENTARY,
    ANY_INT,
    ANY_MAGNITUDE,
    ANY_NUM,
    ANY_REAL,
    ANY_STRING,
    ANY;

    @Override
    public String stringify() {
        return "" + this;
    }
}

File: code/library/src/main/kotlin/org/fbme/lib/st/types/GenericType.kt
package org.fbme.lib.st.types

enum class GenericType : DataType {
    ANY_BIT, ANY_DATE, ANY_DERIVED, ANY_ELEMENTARY, ANY_INT, ANY_MAGNITUDE, ANY_NUM, ANY_REAL, ANY_STRING, ANY;

    override fun stringify(): String {
        return "" + this
    }
}
